gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemNewIntegrationTest.java

```diff
@@ -677,6 +678,83 @@ public void delete_directory_parallel() throws Exception {
     delete_directory(/* parallelStatus= */ true);
   }
 
+  @Test
+  public void delete_inferred_directory() throws Exception {
+    GoogleCloudStorageOptions gcsOptions = GoogleCloudStorageOptions.builder()
+        .setAppName(GoogleCloudStorageTestHelper.APP_NAME)
+        .setAutoRepairImplicitDirectoriesEnabled(false)
+        .setProjectId(checkNotNull(TestConfiguration.getInstance().getProjectId())).build();
```

Comment:
Do we really need to set this?

Reply:
No, we can ensure the asserts accommodate the upload requests